src/client/components/table-of-contents/table-of-contents.vue

```diff
@@ -14,5 +15,78 @@
 <script>
 export default {
   props: ['items'],
+  data () {
+    return {
+      currentSection: this.items[0].slug
+    }
+  },
+  mounted () {
+    window.addEventListener('scroll', this.getClosestSection)
+  },
+  
+  beforeDestroy () {
+    window.removeEventListener('scroll', this.getClosestSection)
+  },
+
+  methods: {
+    getClosestSection () {
+      let sectionsLenght = this.items.length
+
+      for (var i = 0; i < sectionsLenght; i++) {
+        let myElement = document.getElementById(this.items[i].slug)
+
+        if (this.isBelowScroll(myElement)) {
+          this.currentSection = myElement.id
+        }
+      }
+    },
+
+    isBelowScroll (el) {
+      let position = el.getBoundingClientRect()
+      
+      return position.top < 200 && ((position.top + position.height) > 0)
```

Comment:
`200` is a magic value. Add a comment, or put it in a variable with a descriptive name or calculate it so other developers understand where it's coming from.